### ISSUE TITLE
using custom CRS with inverted vertical axis causes multiple problems

### ISSUE THREAD

**Issue by Gjum** [Leaflet/Leaflet#4886]
I use Leaflet 1.0.0-beta.3 with a custom CRS, with transformation `(1, 0, 1, 0)`, which means using a flipped y-coordinate.

When zooming in/out, all image overlays move down before snapping back into place when the zoom ends, so that before snapping back, their top edge is aligned with where their bottom edge was before starting the zoom animation.

Minimal example, based on [the quick start tutorial](http://leafletjs.com/examples/quick-start.html):

http://jsbin.com/conomedolo/edit?html,output

```javascript
        var myCRS = L.extend({}, L.CRS.Simple, {
          transformation: new L.Transformation(1, 0,
            // -1, // works like expected
            1, // image travels while zooming
            0)
        });

        var mymap = L.map('mapid', {crs: myCRS}).setView([0, 0], 3);

        var imageUrl = 'https://avatars2.githubusercontent.com/u/2854298?v=3',
            imageBounds = [[-10, -10], [10, 10]];
        L.imageOverlay(imageUrl, imageBounds).addTo(mymap);
```

**Comment by Gjum** [comment]
I had a look into this and am working around this issue by replacing
```js
offset = this._map._latLngToNewLayerPoint(this._bounds.getNorthWest(), e.zoom, e.center);
```
with
```js
offset = this._map._latLngToNewLayerPoint(this._bounds.getSouthWest(), e.zoom, e.center);
```
in `layer/ImageOverlay.js`, `_animateZoom`, but keep in mind this only works if the y-Axis is inverted.

**Comment by perliedman** [comment]
It looks like the problem is that the code assumes the north western coordinate to be the minimum screen x/y, while for a flipped axis, the minimum y is found from the south western coordinate, like in the workaround posted above.

Since there's no way to really know this without knowing more about the projection, I guess we should really calculate the layer point for all four corners to calculate the minimum x/y. (In the general case, even this approach will not work for all projections, but it will be an improvement).

**Comment by Gjum** [comment]
Something similar happens when using `getBoundsZoom`: I use `L.latLngBounds(c.positions)` to get the bounds for a polygon, but the bounds object stores the vertical coordinates inverted (north > south, although north should be < south in this projection), so the vertical bounds difference is negative and `getScaleZoom` returns `Infinity`.

**Comment by Gjum** [comment]
Thanks for the fix. Should I open another issue for the `getBoundsZoom` bug?

**Comment by perliedman** [comment]
@Gjum I think you should! It sounds a bit trickier, I haven't really wrapped my head around it yet.
